velero-csi-plugin/volumesnapshot_restorer.go

```diff
@@ -38,22 +39,36 @@ func (p *VSRestorer) AppliesTo() (velero.ResourceSelector, error) {
 }
 
 func (p *VSRestorer) Execute(input *velero.RestoreItemActionExecuteInput) (*velero.RestoreItemActionExecuteOutput, error) {
-	p.log.Info("Starting VSRestorer")
+	p.log.Info("Starting VSRestorerAction")
 	var vs snapshotv1beta1api.VolumeSnapshot
 
 	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(input.Item.UnstructuredContent(), &vs); err != nil {
-		return &velero.RestoreItemActionExecuteOutput{}, err
+		return &velero.RestoreItemActionExecuteOutput{}, errors.Wrapf(err, "failed to convert input.Item from unstructured")
+	}
+
+	var vsFromBackup snapshotv1beta1api.VolumeSnapshot
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured((input.ItemFromBackup.UnstructuredContent()), &vsFromBackup); err != nil {
+		return &velero.RestoreItemActionExecuteOutput{}, errors.Wrapf(err, "failed to convert input.ItemFromBackup from unstructured")
 	}
 
-	// This section of code used to set the VolumeSnapshot.Spec.Source field to nil
-	// As of the beta, this is no longer relevant, and the code has been removed.
+	if vsFromBackup.Status == nil || vsFromBackup.Status.BoundVolumeSnapshotContentName == nil {
+		return &velero.RestoreItemActionExecuteOutput{}, errors.Errorf("unable to lookup BoundVolumeSnapshotContentName from status")
+	}
+
+	vscName := *vsFromBackup.Status.BoundVolumeSnapshotContentName
+	vs.Spec = snapshotv1beta1api.VolumeSnapshotSpec{
+		Source: snapshotv1beta1api.VolumeSnapshotSource{
```

Comment:
Is this not already set in the backed-up object? Can you add a comment to clarify the intent of what we're doing here?